src/nbl/builtin/CMakeLists.txt

```diff
@@ -1,293 +1,244 @@
+# virtual geometry
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/virtual_geometry/descriptors.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/virtual_geometry/virtual_attribute.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/virtual_geometry/virtual_attribute_fetch.glsl")
+# depth pyramid generator
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/ext/DepthPyramidGenerator/common.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/ext/DepthPyramidGenerator/push_constants_struct_common.h")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/ext/DepthPyramidGenerator/depth_pyramid_generator_impl.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/ext/DepthPyramidGenerator/virtual_work_group.glsl")
+# blit
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/formats_encode.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/parameters.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/multi_dimensional_array_addressing.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/default_compute_common.comp")
 
-	"nbl/builtin/glsl/blit/default_compute_alpha_test.comp"
-	"nbl/builtin/glsl/blit/alpha_test/alpha_test.glsl"
-	"nbl/builtin/glsl/blit/alpha_test/descriptors.glsl"
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/default_compute_blit.comp")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/blit/blit.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/blit/descriptors.glsl")
 
-	"nbl/builtin/glsl/blit/default_compute_normalization.comp"
-	"nbl/builtin/glsl/blit/normalization/normalization.glsl"
-	"nbl/builtin/glsl/blit/normalization/descriptors.glsl"
-	"nbl/builtin/glsl/blit/normalization/shared_normalization.glsl"
-PARENT_SCOPE)
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/default_compute_alpha_test.comp")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/alpha_test/alpha_test.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/alpha_test/descriptors.glsl")
 
-function(ADD_CUSTOM_BUILTIN_RESOURCES _TARGET_NAME_ _NAMESPACE_ _CUSTOM_RESOURCES_TO_EMBED_ _OUTPUT_DIRECTORY_)
-	set(NBL_BUILTIN_HEADER_GEN_PY "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/builtinHeaderGen.py")
-	set(NBL_BUILTIN_DATA_GEN_PY "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/builtinDataGen.py")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/default_compute_normalization.comp")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/normalization/normalization.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/normalization/descriptors.glsl")
+LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "nbl/builtin/glsl/blit/normalization/shared_normalization.glsl")
```

Comment:
We should list them all without `nbl/builtin` because they're supposed to be relative to the archive (which is either the builtin thing or a mounted directory with absolute path to `ROOT_DIR/nbl/builtin`)

Reply:
resolved